src/init.cpp

```diff
@@ -600,12 +610,14 @@ std::string HelpMessage(HelpMessageMode mode)
     strUsage += HelpMessageOpt("-debugexclude=<category>", strprintf(_("Exclude debugging information for a category. Can be used in conjunction with -debug=1 to output debug logs for all categories except one or more specified categories.")));
     strUsage += HelpMessageOpt("-disablegovernance", strprintf(_("Disable governance validation (0-1, default: %u)"), 0));
     strUsage += HelpMessageOpt("-help-debug", _("Show all debugging options (usage: --help -help-debug)"));
-    strUsage += HelpMessageOpt("-highsubsidyblocks=<n>", _("The number of blocks with a higher than normal subsidy to mine at the start of a devnet (default: 0)"));
-    strUsage += HelpMessageOpt("-highsubsidyfactor=<n>", _("The factor to multiply the normal block subsidy by while in the highsubsidyblocks window of a devnet (default: 1)"));
-    strUsage += HelpMessageOpt("-llmqchainlocks=<quorum name>", _("Override the default LLMQ type used for ChainLocks on a devnet. Allows using ChainLocks with smaller LLMQs. (default: llmq50_60)"));
-    strUsage += HelpMessageOpt("-llmqdevnetparams=<size:threshold>", _("Override the default LLMQ size for the LLMQ_DEVNET quorum (default: 10:6)"));
-    strUsage += HelpMessageOpt("-llmqinstantsend=<quorum name>", _("Override the default LLMQ type used for InstantSend on a devnet. Allows using InstantSend with smaller LLMQs. (default: llmq50_60)"));
-    strUsage += HelpMessageOpt("-llmqtestparams=<size:threshold>", _("Override the default LLMQ size for the LLMQ_TEST quorum (default: 3:2)"));
+    if (gArgs.IsArgSet("-devnet")) {
+        strUsage += HelpMessageOpt("-highsubsidyblocks=<n>", strprintf(_("The number of blocks with a higher than normal subsidy to mine at the start of a devnet (default: %u)"), devnetConsensus.nHighSubsidyBlocks));
+        strUsage += HelpMessageOpt("-highsubsidyfactor=<n>", strprintf(_("The factor to multiply the normal block subsidy by while in the highsubsidyblocks window of a devnet (default: %u)"), devnetConsensus.nHighSubsidyFactor));
+        strUsage += HelpMessageOpt("-llmqchainlocks=<quorum name>", strprintf(_("Override the default LLMQ type used for ChainLocks on a devnet. Allows using ChainLocks with smaller LLMQs. (default: %s)"), devnetConsensus.llmqs.at(devnetConsensus.llmqTypeChainLocks).name));
+        strUsage += HelpMessageOpt("-llmqdevnetparams=<size:threshold>", strprintf(_("Override the default LLMQ size for the LLMQ_DEVNET quorum (default: %u:%u)"), devnetLLMQ.size, devnetLLMQ.threshold));
+        strUsage += HelpMessageOpt("-llmqinstantsend=<quorum name>", strprintf(_("Override the default LLMQ type used for InstantSend on a devnet. Allows using InstantSend with smaller LLMQs. (default: %s)"), devnetConsensus.llmqs.at(devnetConsensus.llmqTypeInstantSend).name));
+    }
+    strUsage += HelpMessageOpt("-llmqtestparams=<size:threshold>", strprintf(_("Override the default LLMQ size for the LLMQ_TEST quorum (default: %u:%u)"), regtestLLMQ.size, regtestLLMQ.threshold));
```

Comment:
I'm not sure I like the idea of running `-devnet="xyz" -help` just to see the help text..

Reply:
Could use `-devnet -?` as a shorter version but yes, it's unfortunate. The reason for wrapping it with `if (gArgs.IsArgSet("-devnet"))` is that creating devnet params triggers `FindDevNetGenesisBlock` which calls `GetDevNetName()` which requires `-devnet` to be set (`assert(IsArgSet("-devnet"));`).

Reply:
Can't we create new consts which aren't a member of devnetConsensus that represents the default values?

Reply:
How about https://github.com/UdjinM6/dash/commit/00f109eaa54e18d165d0d37410235b995b195551 ?

Reply:
I think that's better...